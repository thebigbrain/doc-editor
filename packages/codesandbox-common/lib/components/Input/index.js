"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TextArea = exports.styles = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  transition: 0.3s ease border-color;\n  background-color: ", ";\n  color: ", ";\n  border: none;\n  outline: none;\n  border-radius: 4px;\n  padding: 0.25em;\n  width: ", ";\n  box-sizing: border-box;\n\n  border: 1px solid\n    ", ";\n\n  &:focus {\n    border-color: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var styles = (0, _styledComponents.css)(_templateObject(), function (props) {
  return props.theme['input.background'] || 'rgba(0, 0, 0, 0.3)';
}, function (props) {
  return props.theme['input.foreground'] || (props.theme.light ? '#636363' : 'white');
}, function (_ref) {
  var block = _ref.block,
      fullWidth = _ref.fullWidth;
  return block || fullWidth ? '100%' : 'inherit';
}, function (props) {
  return props.error ? props.theme.red.clearer(0.5) : 'rgba(0, 0, 0, 0.1)';
}, function (props) {
  return props.theme.secondary.clearer(0.6);
});
exports.styles = styles;

var Input = _styledComponents["default"].input(_templateObject2(), styles);

var TextArea = _styledComponents["default"].textarea(_templateObject3(), styles);

exports.TextArea = TextArea;
var _default = Input;
exports["default"] = _default;