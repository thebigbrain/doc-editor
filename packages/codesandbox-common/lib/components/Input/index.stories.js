"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _ = _interopRequireDefault(require("."));

var stories = (0, _react2.storiesOf)('components/Input', module);
stories.add('Basic Input', function () {
  return _react["default"].createElement(_["default"], {
    value: "I am a fancy input"
  });
}).add('Error Input', function () {
  return _react["default"].createElement(_["default"], {
    value: "I am a fancy input",
    error: true
  });
}).add('Placeholder Input', function () {
  return _react["default"].createElement(_["default"], {
    placeholder: "Hello"
  });
});