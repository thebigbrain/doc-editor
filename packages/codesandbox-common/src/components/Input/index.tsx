import styled, { css } from 'styled-components';

export const styles = css<{
  block?: boolean;
  fullWidth?: boolean;
  error?: boolean;
}>`
  transition: 0.3s ease border-color;
  background-color: ${props =>
    props.theme['input.background'] || 'rgba(0, 0, 0, 0.3)'};
  color: ${props =>
    props.theme['input.foreground'] ||
    (props.theme.light ? '#636363' : 'white')};
  border: none;
  outline: none;
  border-radius: 4px;
  padding: 0.25em;
  width: ${({ block, fullWidth }) => (block || fullWidth ? '100%' : 'inherit')};
  box-sizing: border-box;

  border: 1px solid
    ${props =>
      props.error ? props.theme.red.clearer(0.5) : 'rgba(0, 0, 0, 0.1)'};

  &:focus {
    border-color: ${props => props.theme.secondary.clearer(0.6)};
  }
`;

const Input = styled.input`
  ${styles};
`;

export const TextArea = styled.textarea`
  ${styles};
`;

export default Input;
