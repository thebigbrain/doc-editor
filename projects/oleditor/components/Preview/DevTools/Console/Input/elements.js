import styled from 'styled-components';
import { CodeContainer } from '../../../../CodeEditor/CodeMirror/elements';
export const Container = styled.div `
  position: relative;
  height: 2rem;
  min-height: 2rem;
  width: 100%;
  background-color: ${props => props.theme['input.background'] || props.theme.background.darken(0.3)};

  display: flex;
  align-items: center;
`;
export const Input = styled.input `
  position: relative;
  height: 1.5rem;
  width: 100%;
  background-color: ${props => props.theme['input.background'] || props.theme.background.darken(0.3)};
  border: none;
  outline: none;
  color: ${props => props.theme['input.foreground'] || 'rgba(255, 255, 255, 0.8)'};
  font-family: Menlo, monospace;
  font-size: 13px;
`;
export const IconContainer = styled.div `
  display: inline-flex;
  padding: 0.5rem 0;
  width: 24px;
  align-items: center;
  justify-content: center;
`;
export const CodeMirrorContainer = styled(CodeContainer) `
  display: flex;
  align-items: center;
  background-color: ${props => props.theme['input.background'] || props.theme.background.darken(0.3)};

  > div {
    width: 100%;
  }

  .CodeMirror {
    background: ${props => props.theme['input.background'] || props.theme.background.darken(0.3)};
  }
`;
