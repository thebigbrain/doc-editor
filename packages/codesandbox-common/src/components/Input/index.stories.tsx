import React from 'react';
import { storiesOf } from '@storybook/react';
import Input from '.';

const stories = storiesOf('components/Input', module);

stories
  .add('Basic Input', () => <Input value="I am a fancy input" />)
  .add('Error Input', () => <Input value="I am a fancy input" error />)
  .add('Placeholder Input', () => <Input placeholder="Hello" />);
