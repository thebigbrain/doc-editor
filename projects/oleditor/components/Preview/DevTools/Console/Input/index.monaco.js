/* @flow */
import React from 'react'
import styled from 'styled-components'

import ChevronRight from 'react-icons/lib/md/chevron-right'
import theme from '@csb/common/lib/theme'

import MonacoEditor from 'app/components/CodeEditor/Monaco/MonacoReactComponent'
import defineTheme from 'app/components/CodeEditor/Monaco/define-theme'

import { ARROW_DOWN, ARROW_UP, ENTER } from '@csb/common/lib/utils/keycodes'

import { IconContainer } from './elements'

const CONSOLE_INPUT_TOP_PADDING = 6
const CONSOLE_INPUT_BOTTOM_PADDING = 6
const CONSOLE_INPUT_PADDING =
  CONSOLE_INPUT_TOP_PADDING + CONSOLE_INPUT_BOTTOM_PADDING
const CONSOLE_INPUT_LINE_HEIGHT = 20
const CONSOLE_INPUT_MAX_HEIGHT = 110

const Container = styled.div`
  flex-shrink: 0;
  position: relative;
  height: ${props => props.height}px;
  min-height: 2rem;
  max-height: 100%;
  width: 100%;
  background-color: ${props =>
  props.theme['input.background'] || props.theme.background.darken(0.3)};
  display: flex;
  align-items: flex-start;

  .monaco-editor-background {
    background-color: ${props =>
  props.theme['input.background'] || props.theme.background.darken(0.3)};
  }

  .react-monaco-editor-container {
    overflow: visible !important;
  }

  .mtk5 {
    color: #99c794 !important;
  }
  .mtk12.PropertyAssignment {
    color: #99c794;
  }
  .mtk12.PropertyAssignment.PropertyAccessExpression {
    color: #fac863;
  }
  .Identifier.CallExpression
    + .OpenParenToken.CallExpression
    + .Identifier.CallExpression {
    color: #fac863 !important;
  }
`

type
Props = {
  evaluateConsole: (command: string) => void,
}

type
State = {
  editorHeight: number,
  commandHistory: Array < string >,
  commandCursor: number,
}

const InputWrapper = styled.div`
  position: relative;
  height: 100%;
  padding-top: ${CONSOLE_INPUT_TOP_PADDING}px;
  padding-bottom: ${CONSOLE_INPUT_BOTTOM_PADDING}px;
  box-sizing: border-box;
  width: 100%;
`

const monacoOptions = {
  language: 'javascript',
  wordWrap: 'on',
  overviewRulerLanes: 0,
  glyphMargin: false,
  lineNumbers: 'off',
  folding: false,
  selectOnLineNumbers: false,
  selectionHighlight: false,
  cursorStyle: 'line-thin',
  scrollbar: {
    useShadows: false,
    horizontal: 'hidden',
    verticalScrollbarSize: 9,
  },
  lineDecorationsWidth: 0,
  scrollBeyondLastLine: false,
  renderLineHighlight: 'none',
  minimap: {
    enabled: false,
  },
  contextmenu: false,
  ariaLabel: 'ConsoleInput',
  fontFamily: 'dm, Menlo, monospace',
  fontSize: 13,
}

function noop() {
  return Promise.resolve()
}

class ConsoleInput extends React.PureComponent<Props, State> {
  sizeProbeInterval: number

  state = {
    commandHistory: [],
    commandCursor: -1,

    editorHeight: CONSOLE_INPUT_LINE_HEIGHT,
  }

  editor: any

  resizeEditor = () => {
    this.editor.layout()
  }

  editorDidMount = async (editor: any) => {
    this.editor = editor

    let lastLineCount = 1
    editor.onDidChangeModelContent(() => {
      const lineCount = editor.getModel().getLineCount()
      if (lineCount !== lastLineCount) {
        this.setState({
          editorHeight: Math.min(
            CONSOLE_INPUT_MAX_HEIGHT,
            lineCount * CONSOLE_INPUT_LINE_HEIGHT,
          ),
        })
        this.resizeEditor()
        lastLineCount = lineCount
      }
    })

    editor.onKeyDown(event => {
      const e = event.browserEvent
      const { evaluateConsole } = this.props

      if (e.keyCode === ENTER) {
        if (e.shiftKey) {
          return
        }
        e.preventDefault()
        e.stopPropagation()
        const command = editor.getModel().getValue()
        evaluateConsole(command)
        editor.setValue('')
        this.setState(state => ({
          commandCursor: -1,
          commandHistory: [command, ...state.commandHistory],
        }))
      } else if (e.keyCode === ARROW_UP) {
        const { lineNumber } = editor.getPosition()
        if (lineNumber !== 1) {
          return
        }

        this.setState(state => {
          const newCursor = Math.min(
            state.commandCursor + 1,
            state.commandHistory.length - 1,
          )
          editor.setValue(state.commandHistory[newCursor] || '')
          return { commandCursor: newCursor }
        })
      } else if (e.keyCode === ARROW_DOWN) {
        const { lineNumber } = editor.getPosition()
        const lineCount = editor.getModel().getLineCount()
        if (lineNumber !== lineCount) {
          return
        }

        this.setState(state => {
          const newCursor = Math.max(state.commandCursor - 1, -1)
          editor.setValue(state.commandHistory[newCursor] || '')

          return { commandCursor: newCursor }
        })
      }
    })

    window.addEventListener('resize', this.resizeEditor)
    this.sizeProbeInterval = setInterval(this.resizeEditor.bind(this), 3000)
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.resizeEditor)
    clearTimeout(this.sizeProbeInterval)
    if (this.editor) {
      this.editor.dispose()
    }
  }

  render() {
    return (
      <Container height={this.state.editorHeight + 12}>
        <IconContainer style={{ color: theme.secondary() }}>
          <ChevronRight/>
        </IconContainer>
        <InputWrapper>
          <MonacoEditor
            width="100%"
            height={`calc(100% - ${CONSOLE_INPUT_PADDING}px)`}
            options={monacoOptions}
            theme="CodeSandbox"
            editorWillMount={defineTheme}
            editorDidMount={this.editorDidMount}
            openReference={noop}
          />
        </InputWrapper>
      </Container>
    )
  }
}

export default ConsoleInput
