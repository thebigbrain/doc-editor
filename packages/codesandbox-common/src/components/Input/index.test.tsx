import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import Input from '.';

describe('<Input /> rendering', () => {
  it('basic', () => {
    const wrapper = mountWithTheme(<Input value="I am a fancy input" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('error', () => {
    const wrapper = mountWithTheme(<Input value="I am a fancy input" error />);
    expect(wrapper).toMatchSnapshot();
  });
  it('placeholder', () => {
    const wrapper = mountWithTheme(<Input placeholder="Hello" />);
    expect(wrapper).toMatchSnapshot();
  });
});
