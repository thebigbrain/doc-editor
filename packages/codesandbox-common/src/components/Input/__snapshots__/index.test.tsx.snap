// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> rendering basic 1`] = `ReactWrapper {}`;

exports[`<Input /> rendering error 1`] = `ReactWrapper {}`;

exports[`<Input /> rendering placeholder 1`] = `ReactWrapper {}`;
